cmd/traffic_layout/engine.cc

```diff
@@ -281,11 +287,30 @@ RunrootEngine::copy_runroot(const std::string &original_root, const std::string
 
   for (auto it : original_map) {
     std::string old_path = Layout::relative_to(original_root, it.second);
-    std::string new_path = Layout::relative_to(ts_runroot, it.second);
+    std::string new_path;
+    if (relocate_flag) {
+      new_path           = Layout::relative_to(ts_runroot, it.first);
+      path_map[it.first] = Layout::relative_to(".", it.first);
+    } else {
+      new_path           = Layout::relative_to(ts_runroot, it.second);
+      path_map[it.first] = new_path;
+    }
     if (!copy_directory(old_path, new_path)) {
       ink_warning("Copy failed for %s - %s", it.first.c_str(), strerror(errno));
     }
-    path_map[it.first] = new_path;
   }
-  path_map["prefix"] = ts_runroot;
+  // apend the prefix at last
+  if (relocate_flag) {
+    path_map["exec_prefix"] = "./";
+    path_map["prefix"]      = "./";
+    // override the original record config with ENV variables
+    RecConfigOverrideFromEnvironment("proxy.config.config_dir", "syscondir/");
+    RecConfigOverrideFromEnvironment("proxy.config.local_state_dir", "localstatedir/");
+    RecConfigOverrideFromEnvironment("proxy.config.log.logfile_dir", "logdir/");
+    RecConfigOverrideFromEnvironment("proxy.config.bin_path", "bindir/");
+    RecConfigOverrideFromEnvironment("proxy.config.plugin.plugin_dir", "libexecdir/");
```

Comment:
It's pretty unclear to me what you are trying to do here. This will set "proxy.config.plugin.plugin_dir" to either the value of `getenv("PROXY_CONFIG_PLUGIN_PLUGIN_DIR")` or the hard-coded string `"libexecdir/"`. Why would you want to do that?

Reply:
The second one should be nullptr for these functions. This is trying to get the environment variable to overwrite the record config